src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4035,10 +4037,18 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
                 {
                     case SyntaxKind.ThisKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod);
+                        if (seenRefModifier)
+                        {
+                            modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods);
+                        }
                         break;
                     case SyntaxKind.InKeyword:
                     case SyntaxKind.ReadOnlyKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureReadonlyReferences);
+                        seenRefModifier = true;
```

Comment:
If this is meant to handle the ```in``` case, then we should set this flag only for ```in```.

---
In reply to: [123672189](https://github.com/dotnet/roslyn/pull/20403#discussion_r123672189) [](ancestors = 123672189)